src/main/java/com/tabnine/TabNineCompletionContributor.java

```diff
@@ -172,7 +174,12 @@ TabNineProcess getProcOrPrintError() {
                 req.max_num_results = max_num_results;
                 req.region_includes_beginning = (begin == 0);
                 req.region_includes_end = (end == doc.getTextLength());
-                return proc.request(req);
+                
+                try {
+                    return AppExecutorUtil.getAppExecutorService().submit(() -> proc.request(req)).get(1, TimeUnit.SECONDS);
```

Comment:
I think `Application.executeOnPooledThread` is safer (to the best of my knowledge, this is how JB suggesting to run task on pooled thread)

Reply:
![image](https://user-images.githubusercontent.com/69103343/93017488-d6446d00-f5d1-11ea-916e-04cc6e96b2dc.png)
Application is an interface. Did you mean something else? @Eransho

Reply:
I found it and changed it.